pkg/misc/depversion/depversion.go

```diff
@@ -252,27 +252,36 @@ func parseSemver(s string) (semver, major, minor, patch, prerelease, metadata st
 }
 
 func parseSemverHelper(re *regexp.Regexp, s string) (semver, major, minor, patch, prerelease, metadata string, err error) {
-	matches := re.FindStringSubmatch(s)
-
-	if len(matches) == 0 {
-		err = fmt.Errorf("Did not match regex: %q %s", s, re)
-		return
-	}
-	semverIdx := re.SubexpIndex("semver")
-	majorIdx := re.SubexpIndex("major")
-	minorIdx := re.SubexpIndex("minor")
-	patchIdx := re.SubexpIndex("patch")
-	prereleaseIdx := re.SubexpIndex("prerelease")
-	metadataIdx := re.SubexpIndex("metadata")
-
-	if semverIdx < 0 {
-		err = fmt.Errorf("unable to find semver")
-		return
-	}
+	tildeRegex := regexp.MustCompile(`^(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>[xX])$`)
```

Comment:
i'm not sure i see the tilde in this regex?

Reply:
This is my first time writing regex, so I had some issues. I believe that I've managed to rectify all problems, but please point out any errors you might come across.

***

Previously when `s` hit line `337` (Currently, outside of this commit it is line `334`) it would match the regex, but with `s` having a tilde and a wildcard, it wouldn't. It would continue until it hit line `370` (Currently, outside of this commit it is line `367`) where it would check for a tilde. The issue with this is that `s` has a wildcard, but is skipping the wildcard check.

To remedy this I decided to update `isSemVerWildcard`. 

In order to do this I have created a new regex that checks for a tilde and than a `".x"` anywhere else in the string:

``` go
var exactSvRWithWildcardAndTidle = regexp.MustCompile(`~.*[.]x`) 
```

Then inside the function `isSemVerWildcard` I am checking `exactSvRWithWildcardAndTidle || exactSvRWithWildcard`. The `exactSvRWithWildcardAndTidle` checks cases that have a tilde AND a `"x"` (ie. `"~0.10.x"`), while `exactSvRWithWildcard` checks cases that only have a wildcard (ie. `"0.10.x"`). 

Note that `exactSvRWithWildcardAndTidle` won't be checking cases with just a tilde and no wildcard, `"~0.10.0"`. 

***

I also removed everything I did to `parseSemverHelper` because I am now just trimming off the beginning tilde.

Reply:
The regex has been removed because they weren't very necessary https://github.com/guacsec/guac/pull/1157#pullrequestreview-1581436663. Instead the code is now ignoring the tilde when checking for a wildcard.